File: investment-simulator-api/src/InvestmentSimulator.Domain/Entities/Contribution.cs
namespace InvestmentSimulator.Domain.Entities;

/// <summary>
/// A single contribution (aporte) with date and amount (ERS section 4).
/// Monetary values use <see cref="decimal"/> (ERS section 28).
/// </summary>
public sealed class Contribution
{
    public Contribution(DateOnly date, decimal amount)
    {
        Date = date;
        Amount = amount;
    }

    /// <summary>Contribution date.</summary>
    public DateOnly Date { get; }

    /// <summary>Contribution amount in BRL.</summary>
    public decimal Amount { get; }
}

File: investment-simulator-ui/src/types/contribution.ts
/**
 * Additional contribution (aporte) from ERS section 4.
 * Monetary values are kept as strings to avoid float precision loss.
 */
export type ContributionInput = {
  /** Contribution date (YYYY-MM-DD). */
  date: string;
  /** Contribution amount in BRL as a decimal string (e.g. "1000.50"). */
  amount: string;
};

/**
 * Row used by the dynamic contributions table (includes a stable client id).
 */
export type ContributionRow = ContributionInput & {
  id: string;
};

export type ContributionRowErrors = {
  date?: string;
  amount?: string;
};

export type ContributionsErrors = Record<string, ContributionRowErrors>;
